share/spice/currency/currency.js

```diff
@@ -163,15 +459,70 @@
             },
             templates: templateObj,
             onShow: function() {
-                // The desktop template depends on a JS function that manages the
-                // size of the container.
-                if(!is_mobile) {
-                    $(window).on('load', resize);
-                    $(window).resize(resize);
-                } else {
-                    $(window).on('load', resizeMobile);
-                    $(window).resize(resizeMobile);
+
+                if(!initialized) {
+                    var $currency = $("#zci-currency");
+                    $flag_image_right = $currency.find("#flag-img-right");
+                    $flag_image_left = $currency.find("#flag-img-left");
+                    $currency_input_left = $currency.find("#zci--currency-amount-left");
+                    $currency_input_right = $currency.find("#zci--currency-amount-right");
+                    $left_select = $currency.find("select#zci--currency-symbol-left");
+                    $right_select = $currency.find("select#zci--currency-symbol-right");
+                    $selects = $currency.find("select.zci--currency-symbol");
+                    $more_at_link_normal = $currency.find(".zci__more-at");
+                    $more_at_link_charts = $currency.find(".zci__more-at--info");
+                    $to_from_label = $currency.find("#to-from-label");
+                    $from_to_label = $currency.find("#from-to-label");
+
+                    // apends all the currency names to the selects
+                    for( var i = 0 ; i < currencies.length ; i ++ ) {
+                        $selects.append(
+                            "<option value=" + 
+                            currencies[i] + 
+                            ">" + 
+                            currencies[i] +
+                            "</option>"
+                        );
+                    }
+
+                    // sets the default
+                    $left_select.val(Converter.from_currency);
+                    $right_select.val(Converter.to_currency);
+                }
+
+                /**
+                 * When the user clicks on the field we select
```

Comment:
Not sure how I feel about this functionality. It's pretty awkward when I want to go to insert a character; I have to go to my keyboard and hit an arrow key to do it.
![ddgtemp](https://cloud.githubusercontent.com/assets/4411471/26223881/13272f38-3bee-11e7-90ff-3564f6d3bf7a.gif)